lib/protoparser/statsd/parser.go

```diff
@@ -5,15 +5,47 @@ import (
 	"strings"
 
 	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/slicesutil"
 	"github.com/VictoriaMetrics/metrics"
 	"github.com/valyala/fastjson/fastfloat"
 )
 
 // Statsd metric format with tags: MetricName:value|type|@sample_rate|#tag1:value,tag1...
-const statsdSeparator = '|'
-const statsdPairsSeparator = ':'
-const statsdTagsStartSeparator = '#'
-const statsdTagsSeparator = ','
+// https://docs.datadoghq.com/developers/dogstatsd/datagram_shell?tab=metrics#the-dogstatsd-protocol
+const (
+	statsdSeparator          = '|'
+	statsdPairsSeparator     = ':'
+	statsdTagsStartSeparator = '#'
+	statsdTagsSeparator      = ','
+	statsdTypeTagName        = "__statsd_metric_type__"
+)
+
+// https://github.com/b/statsd_spec
+var validTypes = []string{
+	// counter
+	"c",
+	// gauge
+	"g",
+	// histogram
+	"h",
+	// timer
+	"ms",
+	// distribution
+	"d",
+	// set
+	"s",
+	// meters
+	"m",
+}
+
+func isValidType(src string) bool {
```

Comment:
Good point. I think, docs must be updated with a list of supported metric types.

And for `Not Assigned` type, it could bring more harm than good. By ignoring this type, we force users to fix their clients instead of fixing data inside database(update operation required for this label, maybe vmalert replay or smth like that) and possible performance issues with it ( if this type bypass streaming aggregation matcher, ingest load could be increased x10-20 time).